src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxList.WithLotsOfChildren.cs

```diff
@@ -53,7 +53,19 @@ public override int FindSlotIndexContainingOffset(int offset)
             {
                 Debug.Assert(offset >= 0 && offset < FullWidth);
                 int idx = _childOffsets.BinarySearch(offset);
-                return idx >= 0 ? idx : (~idx - 1);
+
+                if (idx < 0)
+                {
+                    idx = (~idx - 1);
```

Comment:
I think you can return here for this case. i.e. If the offset isn't found, then just return ~idx - 1;
You only need to look closer if you get a 'hit'.